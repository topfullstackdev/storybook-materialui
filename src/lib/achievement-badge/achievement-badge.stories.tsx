import { Meta, Story } from '@storybook/react';

import { AchievementBadge, AchievementBadgeProps } from './achievement-badge';
import exampleImage from './example-badge.png';

export default {
  component: AchievementBadge,
  title: 'AchievementBadge',
  decorators: [],
  args: {
    name: 'Example Badge',
    src: exampleImage,
  },
} as Meta;

const Template: Story<AchievementBadgeProps> = function Template(
  props: AchievementBadgeProps,
) {
  return <AchievementBadge {...props} />;
};

export const WithoutName = Template.bind({});
WithoutName.args = {
  showName: false,
};
export const WithName = Template.bind({});
WithName.args = {
  showName: true,
};
