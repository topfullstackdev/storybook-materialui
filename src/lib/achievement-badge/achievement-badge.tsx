import styled from '@emotion/styled';

import { Text } from '../text/text';

export interface AchievementBadgeProps {
  src: string;
  name: string;
  showName?: boolean;
}

const SmallContainer = styled.div`
  text-align: left;
`;
const SmallImage = styled.img`
  height: 48px;
  vertical-align: middle;
`;

const LargeContainer = styled.div`
  text-align: center;
`;
const LargeImage = styled.img`
  width: 100px;
  height: 100px;
  margin-bottom: 5px;
`;
const Name = styled.div`
  width: 100%;
`;

export function AchievementBadge({
  name,
  showName,
  src,
}: AchievementBadgeProps) {
  const imgProps = { src, alt: name };
  return showName ? (
    <LargeContainer>
      <LargeImage {...imgProps} data-cy={`CP Badge Image:${name}`} />
      <Name data-cy={`CP Name:${name}`}>
        <Text variant="caption">{name} </Text>
      </Name>
    </LargeContainer>
  ) : (
    <SmallContainer>
      <SmallImage {...imgProps} />
    </SmallContainer>
  );
}

export default AchievementBadge;
