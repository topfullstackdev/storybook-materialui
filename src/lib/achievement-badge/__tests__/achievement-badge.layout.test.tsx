import { render } from '@testing-library/react';

import { AchievementBadge } from '../achievement-badge';
import exampleImage from '../example-badge.png';

describe('AchievementBadge', () => {
  it('should render successfully', () => {
    const { container } = render(
      <AchievementBadge name="Example" src={exampleImage} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with name', () => {
    const { container } = render(
      <AchievementBadge name="Example" src={exampleImage} showName />,
    );
    expect(container).toMatchSnapshot();
  });
});
