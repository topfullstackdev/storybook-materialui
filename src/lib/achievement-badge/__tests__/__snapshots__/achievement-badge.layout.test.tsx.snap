// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchievementBadge should render successfully 1`] = `
.emotion-0 {
  text-align: left;
}

.emotion-2 {
  height: 48px;
  vertical-align: middle;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <img
      alt="Example"
      class="emotion-2 emotion-3"
      src="__mock__/fileMock.png"
    />
  </div>
</div>
`;

exports[`AchievementBadge should render with name 1`] = `
.emotion-0 {
  text-align: center;
}

.emotion-2 {
  width: 100px;
  height: 100px;
  margin-bottom: 5px;
}

.emotion-4 {
  width: 100%;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.66;
  letter-spacing: 0.03333em;
  font-weight: normal;
  font-style: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Verdana,Arial,sans-serif;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <img
      alt="Example"
      class="emotion-2 emotion-3"
      data-cy="CP Badge Image:Example"
      src="__mock__/fileMock.png"
    />
    <div
      class="emotion-4 emotion-5"
      data-cy="CP Name:Example"
    >
      <span
        class="MuiTypography-root MuiTypography-caption emotion-6 emotion-7"
      >
        Example
         
      </span>
    </div>
  </div>
</div>
`;
